client/src/components/FlashCard/CreateFlashcard.js

```diff
@@ -0,0 +1,108 @@
+import React, {useState} from "react"
+import { Button, Stack, TextField } from "@mui/material"
+import axios from 'axios'
+
+const CreateFlashcard = ({ userId, deckId }) => {
+  // how can we use state here to make sure we're validating info
+  console.log(`[CreateFlashcard] deckId is ${deckId}`)
+  const [formValue, setFormValue] = useState({})
+  const [errors, setErrors] = useState({
+    'frontImage': false,
+    'frontText': false,
+    'backImage': false,
+    'backText': false
+
+  })
+
+  function validateProperty(fieldName, fieldValue) {
+    console.log('validateProperty', typeof fieldValue)
+    const fieldValueTrimmed = fieldValue.trim()
+    if (fieldValueTrimmed === '') {
+      formValue.frontImage = ''
+      return false
+    } else {
+      return true
+    }
+  }
+
+  const handleChange = (event) => {
+    event.preventDefault()
+    console.log("[CreateFlashcard] onChange ", event)
+    if (validateProperty(event.target.name, event.target.value)) {
+      const currentValues = formValue
+      currentValues[event.target.name] = event.target.value
+      setFormValue(currentValues)
+    } else {
+      return true
+    }
+  }
+  
+  const handleSubmit = async (event) => {
+    console.log("[CreateFlashcard] onSubmit ", event)
+    event.preventDefault()
+    if(formValue.frontImage === '' || formValue.frontText === '' || formValue.backImage === '' || formValue.backText === ''){
+      alert("data is invalid")
```

Comment:
An alert is a very blunt way to communicate invalid form state. What are other, more user-friendly ways to communicate this.